src/QuantQueue.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+namespace Drupal\quant;
+
+use Drupal\Core\Database\IntegrityConstraintViolationException;
+use Drupal\Core\Queue\DatabaseQueue;
+use Drupal\Core\Site\Settings;
+
+/**
+ * Additional handling for Quant queue items.
+ *
+ * @see https://git.drupalcode.org/project/queue_unique
+ */
+class QuantQueue extends DatabaseQueue {
+  /**
+   * The table name to store the queue items.
+   */
+  public const TABLE_NAME = 'quant_queue';
+
+  /**
+   * Length of the hash coulmn.
+   */
+  public const HASH_LENGTH = 48;
+
+  /**
+   * The algorithm used to hash the item.
+   */
+  public const HASH_METHOD = 'sha2';
```

Comment:
`sha2` used here and `sha512` elsewhere. Okay?

Reply:
This actually isn't really used anywhere — you can't reverse the sha's this works by being the same as the previous entry. But let's make them consistent so there's no confusion here.